### DISCUSSION TITLE
How to log the output of the Git commands?

### DISCUSSION THREAD

**Discussion by kschroeer** [czproject/git-php#79] [Q&A] [answered]
Is there a way to log the output of the Git commands?

I didn't find any function right away that expected a LoggerInterface as a parameter, for example.

I noticed that the outputs from StdOut and StdErr are captured in the RunnerResult object. That would actually be exactly what I need.
But unfortunately only the run() method returns the RunnerResult object, not the commit() and push() methods.
So when I commit, I no longer have a way to access the RunnerResult.
How do I continue here?

**Comment by janpecha** [comment, marked as answer]
Hello. You can create own implementation of IRunner, something like:

```php
class LoggingRunner implements \CzProject\GitPhp\IRunner
{
	private $logger;
	private $runner;


	public function __construct(
		LoggerInterface $logger,
		\CzProject\GitPhp\IRunner $runner
	)
	{
		$this->logger = $logger;
		$this->runner = $runner;
	}


	public function run($cwd, array $args, array $env = NULL)
	{
		$result = $this->runner->run($cwd, $args, $env);

		// log result
		$this->logger->log($result);

		return $result;
	}


	public function getCwd()
	{
		return $this->runner->getCwd();
	}
}


$git = new \CzProject\GitPhp\Git(
	new LoggingRunner(
		$logger,
		new \CzProject\GitPhp\Runners\CliRunner
	)
);
$repo = $git->open('/path/to/repo');
```
